apps/api/src/lib/web-of-knowledge/decorators/invariant-validation-method.decorator.ts

```diff
@@ -0,0 +1,33 @@
+import validateSimpleInvariants from '../../../domain/domainModelValidators/utilities/validateSimpleInvariants';
+import { isValid, Valid } from '../../../domain/domainModelValidators/Valid';
+import { Aggregate } from '../../../domain/models/aggregate.entity';
+import { InternalError } from '../../errors/InternalError';
+import { ValidationResult } from '../../errors/types/ValidationResult';
+
+type ErrorFactoryFunction = (innerErrors: InternalError[], instance: Aggregate) => InternalError;
+
+export function InvariantValidationMethod(
+    errorFactoryFunction: ErrorFactoryFunction
+): MethodDecorator {
+    return function (target: object, _: string | symbol, descriptor: PropertyDescriptor) {
+        const childFunction = descriptor.value;
+
+        descriptor.value = function () {
+            const simpleValidationResult = validateSimpleInvariants(
+                (target as any).constructor,
```

Comment:
I just pushed the any to the top and left it. Type safety doesn't do us a lot of good here, as this is our step out of the static TypeScript world and into the dynamic JavaScript world. The only thing we could look into is forcing the decorator to be used on a method on a class that extends resource or maybe having an automated test that check this at run time. We can do that when we hit the problem. Otherwise, just use the decorator properly!